### PR TITLE
as branch name

### PR DESCRIPTION
**Pull request by victorjarlow**

### CODE REVIEW

**Review by konglobemeralt — COMMENTED** [review]
This is a good fix for the issue that we are having right now, but was the issue not that arm did not like the htol function? 

I think a better way would be to check if we are building for arm or something else. I will look into this.

**Comment by konglobemeralt** [comment]
Hmm looks like most ways are a bit "hacky" 
image
Maybe leave as is for now

**Review by konglobemeralt — APPROVED** [review]

**Comment by victorjarlow** [comment]
> Hmm looks like most ways are a bit "hacky" image Maybe leave as is for now

Is it ever

**Comment by victorjarlow** [comment]
> Hmm looks like most ways are a bit "hacky" image Maybe leave as is for now

That solution only works when host and target system are the same, since we want to cross compile on our host systems (x86_64) to the target system (maybe arm or others), this would not work.

**Comment by victorjarlow** [comment]
A commenter in the SO thread in the pic: https://stackoverflow.com/questions/11944060/how-to-detect-target-architecture-using-cmake suggests that CMAKE_SYSTEM_PROCESSOR contains the arch of the target system. However, the OP is complaining that CMAKE_SYSTEM_PROCESSOR gives the wrong arch when compiling with a mac host system. 

When I tried printing CMAKE_SYSTEM_PROCESSOR cmake correctly output aarch64 (i.e. arm64) when using my x86_64 host building for target arm64, so this might be a better way forward!

File: CMakeLists.txt [hunk 1]
```diff
@@ -54,18 +54,22 @@ install(TARGETS ${ISO22133_TARGET}
...
-)
-target_include_directories(${ISO22133_TARGET}_test PUBLIC
-	${CMAKE_CURRENT_SOURCE_DIR}
-	${CMAKE_CURRENT_SOURCE_DIR}/include
-)
-include(GoogleTest)
-gtest_discover_tests(${ISO22133_TARGET}_test)
+
+#only build tests if swig is not used upstream
+if ("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64" AND (NOT "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "${CMAKE_HOST_SYSTEM_PROCESSOR})") #target is x86_64 arch and we are not cross compling
```
Comment (victorjarlow) [inline CMakeLists.txt:59]:
```suggestion
if ("${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "x86_64" AND (NOT "${CMAKE_SYSTEM_PROCESSOR}" STREQUAL "${CMAKE_HOST_SYSTEM_PROCESSOR}")) #target is x86_64 arch and we are not cross compling
```

**Pull request merged**